docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -1055,8 +1055,79 @@ Because `MediaLibrary` is the `TvLibrary` class parent, a user with the `admin`
 
 CAUTION: Annotation-based permissions do not work with custom xref:security-customization.adoc#jaxrs-security-context[Jakarta REST SecurityContexts] because there are no permissions in `jakarta.ws.rs.core.SecurityContext`.
 
+[[permission-checker]]
+==== Create permission checkers
+
+By default, `SecurityIdentity` must be augmented with permissions which can be used to check if this identity passes `@PermissionAllowed` authorization restrictions.
+Alternatively, you can use a `@PermissionChecker` annotation to mark any CDI bean method as a permission checker.
+For example, a permission checker can be created like this:
+
+[source,java]
+----
+package org.acme.security.rest.resource;
+
+import io.quarkus.security.PermissionChecker;
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.security.identity.SecurityIdentity;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+
+import org.jboss.resteasy.reactive.RestForm;
+import org.jboss.resteasy.reactive.RestPath;
+
+@Path("/project")
+public class ProjectResource {
+
+    @PermissionsAllowed("rename-project")
+    @POST
+    public void renameProject(@RestPath String projectName, @RestForm String newName) {
+        Project project = Project.findByName(projectName);
+        project.name = newName;
+    }
+
+    @PermissionChecker("rename-project") <2>
+    boolean canRenameProject(SecurityIdentity identity, String projectName) { <3>
+        var principalName = identity.getPrincipal().getName();
+        var user = User.getUserByName(principalName);
+        return userOwnsProject(projectName, user);
+    }
+}
+----
+<1> Permission required to access the `ProjectResource#renameProject` is the `rename-project` permission.
+<2> The `ProjectResource#canRenameProject` method authorize access to the `ProjectResource#renameProject` endpoint.
+<3> The `SecurityIdentity` instance can be injected into any permission checker method.
+
+The permission checker above requires the `SecurityIdentity` instance to authorize the `renameProject` endpoint.
+Instead of declaring the `rename-project` permission checker directly on the resource, you can declare it on any CDI bean like in this example:
+
+[source,java]
+----
+package org.acme.security.rest.resource;
+
+import io.quarkus.security.PermissionChecker;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped <1>
+public class ProjectPermissionChecker {
+
+    @PermissionChecker("rename-project")
+    boolean canRenameProject(String projectName, SecurityIdentity identity) {   <2>
+        var principalName = identity.getPrincipal().getName();
+        var user = User.getUserByName(principalName);
+        return userOwnsProject(projectName, user);
+    }
+
+}
+----
+<1> A CDI bean with the permission checker must be either a normal scoped bean or a `@Singleton` bean.
+<2> The permission checker method must return either `boolean` or `Uni<Boolean>`. Private checker methods are not supported.
```

Comment:
How are failures handled?

Reply:
Failures result in forbidden exception [because only authenticated identities are authorized by these methods). Just FYI, there is a test for that. Exceptions thrown by the check method are logged, but are wrapped.